src/Envelope.cpp

```diff
@@ -302,25 +302,23 @@ static double Limit( double Lo, double Value, double Hi )
 }
 #endif
 
-bool Envelope::HandleXMLTag(const wxChar *tag, const wxChar **attrs)
+bool Envelope::HandleXMLTag(const std::string_view& tag, const AttributesList& attrs)
 {
    // Return unless it's the envelope tag.
-   if (wxStrcmp(tag, wxT("envelope")))
+   if (tag != "envelope")
       return false;
 
-   int numPoints = 0;
-   long nValue = -1;
+   int numPoints = -1;
 
-   while (*attrs) {
-      const wxChar *attr = *attrs++;
-      const wxChar *value = *attrs++;
-      if (!value)
-         break;
-      const wxString strValue = value;
-      if( !wxStrcmp(attr, wxT("numpoints")) &&
-            XMLValueChecker::IsGoodInt(strValue) && strValue.ToLong(&nValue))
```

Comment:
A general comment about these XMLValueChecker functions:  some become unused, so add a commit to remove them.

Do they all become unused?  If they don't, maybe you missed one of the needed rewrites.

In each place where one of the calls to these functions was eliminated, I want to review again to make sure that some other validation is there, with TryGet() not Get().

Reply:
Good point!

Reply:
I found nothing wrong.

Only this:  two tests of in WaveClip and CommandManager, where it's not clear to me why you substituted that for IsGoodString there but not elsewhere.

But IsGoodString, we agree, was a check we just didn't need.